.travis.yml

```diff
@@ -11,95 +12,131 @@ git:
   - helm repo update
-  - ./dev upgrade --chart jupyterhub/jupyterhub --version 0.8.2
-  - ./dev upgrade --chart ./jupyterhub
-  - ./dev port-forward
-  - ./dev test
+
+  # Install a local ACME server
+  - helm install pebble jupyterhub/pebble --values dev-config.yaml
+
+  # Build our images if needed and update values.yaml with the tags
+  - chartpress
+
+  # Pre-install JupyterHub
+  - await_dns
+  - await_pebble
+script:
+  # Install JupyterHub
+  - helm install jupyterhub ./jupyterhub --values dev-config.yaml
+  - await_jupyterhub
+  - await_tls_cert_acquisition
+
+  # Run tests
+  - pytest --verbose --exitfirst ./tests
+
 env:
-  ## NOTE: The environment variables will be expanded to multiple jobs. For
-  ##       additional individual jobs, only the first entry is used.
-  ##
-  ## ref: https://docs.travis-ci.com/user/build-stages/#build-stages-and-build-matrix-expansion
-  ##
-  ##
-  ## Z2JH_KUBE_VERSION should match a released kindest/node image tag, but they
-  ## are currently not automatically published.
-  ##
-  ## ref: https://hub.docker.com/r/kindest/node/tags
-  ## ref: https://github.com/kubernetes-sigs/kind/issues/197
-  ##
-  - Z2JH_KUBE_VERSION=1.13.12
-  - Z2JH_KUBE_VERSION=1.15.6
-  - Z2JH_KUBE_VERSION=1.16.3
-  - Z2JH_KUBE_VERSION=1.16.3 Z2JH_HELM_VERSION=3.0.2
+  global:
+    - K3S_VERSION=v1.18.2+k3s1
+    - HELM_VERSION=v3.2.1
 
 jobs:
-  ## allow experimental setups to fail
-  ##
-  ## ref: https://docs.travis-ci.com/user/customizing-the-build/#rows-that-are-allowed-to-fail
-  ##
-  allow_failures:
-    - name: check docs
   ## don't wait for the jobs that are allowed to fail to report success
   ##
+  ## ref: https://docs.travis-ci.com/user/customizing-the-build/#rows-that-are-allowed-to-fail
   ## ref: https://docs.travis-ci.com/user/customizing-the-build/#fast-finishing
   ##
+  allow_failures:
+    - name: docs:link-check
   fast_finish: true
 
-  ## include additional individual jobs
-  ##
+  ## define individual jobs that can rely on defaults from above
+  ## k3s versions:  https://github.com/rancher/k3s/releases
+  ## helm versions: https://github.com/helm/helm/releases
   include:
-    - stage: lint and validate
+    - stage: test
+      name: chart:k8s-1.18
+
+    - stage: test
+      name: chart:k8s-1.17
+      env: K3S_VERSION=v1.17.5+k3s1 HELM_VERSION=v3.2.1
+
+    - stage: test
+      name: chart:k8s-1.16
+      env: K3S_VERSION=v1.16.9+k3s1 HELM_VERSION=v3.2.1 k3s_disable_command=--no-deploy
+
+    - stage: test
+      name: chart:install-latest-then-upgrade
       script:
+        ## Install latest released JupyterHub
+        ## NOTE: For 0.9.0 we have disabled some parts here because they would
+        ##       require custom configuration that isn't needed post 0.9.0.
+        ##       This regards proxy.traefik.extraVolumes etc and more complete
+        ##       network policies that contain more of whats required by
+        ##       default.
```

Comment:
```suggestion
        ## NOTE: To use Pebble in TravisCI, we require the ability to mount a
        ##       certificate on the autohttps pod, but we can't because that
        ##       configuration ability was added after 0.9.0. Due to this
        ##       autohttps is disabled here.
        ##       The current dev-config.yaml require network policies to contain
        ##       more pre-defined defaults to reach the DNS server not present
        ##       in 0.9.0, and are due to this disabled.
```